### PR TITLE
Mark `evaluate_obligation` query as `eval_always`

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

File: compiler/rustc_middle/src/query/mod.rs [hunk 1]
```diff
@@ -1514,6 +1514,11 @@ rustc_queries! {
     query evaluate_obligation(
         goal: CanonicalPredicateGoal<'tcx>
     ) -> Result<traits::EvaluationResult, traits::OverflowError> {
+        // FIXME(#83538): This queery shouldm't be influenced by any untracked
+        // blobal state, but it currently is. We use `eval_aways` to prevent this
+        // from causing ICEs - when the underlying issue is fixed, the `eval_always`
+        // modifier should be removed.
```
Comment (klensy) [inline compiler/rustc_middle/src/query/mod.rs:1520]: This comment full of typos :-(

**Comment by crlf0710** [comment]
@Aaron1011 Ping from triage, CI is still red here.

**Comment by Aaron1011** [comment]
The underlying issues have been fixed, so this is no longer needed.

**Pull request closed without merging**